types/types.go

```diff
@@ -0,0 +1,73 @@
+package types
+
+import (
+	"log"
+	"sync"
+)
+
+// Runner is an interface which is implemented by all proxies
+type Runner interface {
+	Run() error
+	Cancel() error
+	Reload() error
+}
+
+// Server governs a slice of Runners
+type Server struct {
+	Servers []Runner
+	wg      sync.WaitGroup
+	Done    chan error
+}
+
+func (s *Server) Run() error {
+	var err error
+	for i := range s.Servers {
+		s.wg.Add(1)
+		go func(i int) {
+			defer s.wg.Done()
+			err = s.Servers[i].Run()
+			if err != nil {
+				s.Done <- err
```

Comment:
this means that if all `Servers` end without error, we will never be _done_....

Reply:
Pretty much that means that we must keep track of the number of servers, something that I wanted to avoid :-/